mgmtd/mgmt_txn.c

```diff
@@ -2820,22 +2823,32 @@ int mgmt_txn_notify_rpc_reply(struct mgmt_be_client_adapter *adapter,
 
 	rpc = txn_req->req.rpc;
 
-	/* we don't expect more than one daemon to provide output for an RPC */
-	if (!rpc->client_results && data_len > 0) {
+	tree = NULL;
+	if (data_len)
 		err = yang_parse_rpc(rpc->xpath, reply_msg->result_type,
-				     reply_msg->data, true,
-				     &rpc->client_results);
+				     reply_msg->data, true, &tree);
+	if (err) {
```

Comment:
treating `err` as a boolean - is that safe, long-term? would it be clearer to test against `LY_SUCCESS` explicitly?

Reply:
This is a pretty well trodden pattern, so I'd say it's safe. I actually prefer the code reading "if error" rather than "if error equals success" which feels klunky to me.